modcc/kinrewriter.cpp

```diff
@@ -0,0 +1,159 @@
+#include <iostream>
+#include <map>
+#include <string>
+#include <list>
+
+#include "astmanip.hpp"
+#include "symdiff.hpp"
+#include "visitor.hpp"
+
+class KineticRewriter : public BlockRewriterBase {
+public:
+    using BlockRewriterBase::visit;
+
+    KineticRewriter() {}
+    KineticRewriter(scope_ptr enclosing_scope): BlockRewriterBase(enclosing_scope) {}
+
+    virtual void visit(ConserveExpression *e) override;
+    virtual void visit(ReactionExpression *e) override;
+
+protected:
+    virtual void reset() override {
+        BlockRewriterBase::reset();
+        dterms.clear();
+    }
+
+    virtual void finalise() override;
```

Comment:
Will fix. Do we (or CSCS) have a policy on standard spelling? I'd be quite happy with e.g. Oxford spelling — it's pretty much the international standard, and preferences _-ize_ endings where etymologically justified.